src/data/caniuse.rs

```diff
@@ -1,31 +1,38 @@
 use ahash::AHashMap;
-use std::{borrow::Cow, sync::LazyLock};
+use std::{
+    borrow::{Borrow, Cow},
+    fmt,
+    sync::LazyLock,
+};
 
 pub(crate) mod features;
 pub(crate) mod region;
 
+use crate::data::utils::BinMap;
+
 pub const ANDROID_EVERGREEN_FIRST: f32 = 37.0;
 pub const OP_MOB_BLINK_FIRST: u32 = 14;
 
 #[derive(Clone, Debug)]
-pub struct BrowserStat {
-    name: &'static str,
-    pub version_list: Vec<VersionDetail>,
-}
+pub struct BrowserStat(u32, u32);
+
+#[derive(Clone, Copy)]
+pub struct PooledStr(u32);
 
 #[derive(Clone, Debug)]
 pub struct VersionDetail {
-    pub version: &'static str,
+    pub version: PooledStr,
+    pub release_date: i64,
+    // Use bool instead of Option to use pad space
+    pub released: bool,
     pub global_usage: f32,
-    pub release_date: Option<i64>,
 }
 
-pub type CaniuseData = AHashMap<&'static str, BrowserStat>;
+include!("../generated/caniuse-browsers.rs");
 
-pub static CANIUSE_BROWSERS: LazyLock<CaniuseData> =
-    LazyLock::new(|| include!("../generated/caniuse-browsers.rs"));
+pub static CANIUSE_BROWSERS: BinMap<PooledStr, BrowserStat> = BinMap(BROWSERS_STATS);
 
-pub static CANIUSE_GLOBAL_USAGE: &[(&'static str, &'static str, f32)] =
+pub static CANIUSE_GLOBAL_USAGE: &[(PooledStr, PooledStr, f32)] =
     include!("../generated/caniuse-global-usage.rs");
 
 pub static BROWSER_VERSION_ALIASES: LazyLock<
```

Comment:
Well, we are not completely static, we still have two legacy `LazyLock`. but their data size is not large, so the memory usage is acceptable.